op-node/cmd/batch_decoder/reassemble/reassemble.go

```diff
@@ -107,8 +107,12 @@ func processFrames(cfg Config, rollupCfg *rollup.Config, id derive.ChannelID, fr
 		}
 	}
 
-	var batches []derive.Batch
-	var batchTypes []int
+	var (
+		batches    []derive.Batch
+		batchTypes []int
+		comprAlgos []derive.CompressionAlgo
```

Comment:
why are we building a list of algos, wouldn't there only ever be one at this point, when processing frames? I guess theoretically there could be a change in algo, but I hadn't expected the decoder to survive that.

Reply:
I tend to think so as well, particularly if we are only looking for frames with the same channel id

Reply:
I just followed the pattern for `batchTypes` as well. There usually should ever only be one. But this is a debugging tool, so if there ever are different ones, we can detect it here.